features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java

```diff
@@ -87,15 +92,32 @@
         final Map<String, Object> params = new HashMap<>();
         final String configType = getKeyedString(parameters, CONFIG_TYPE, ConfigType.Default);
         final OnmsIpInterface ipInterface = ipInterfaceDao.findByNodeIdAndIpAddress(svc.getNodeId(), svc.getIpAddr());
-        final Optional<org.opennms.features.deviceconfig.persistence.api.DeviceConfig> deviceConfigOptional = deviceConfigDao.getLatestConfigForInterface(ipInterface, configType);
+        final var deviceConfigOptional = deviceConfigDao.getLatestConfigForInterface(ipInterface, configType);
+        deviceConfigOptional.ifPresent(deviceConfig -> params.put(LAST_RETRIEVAL, String.valueOf(deviceConfig.getLastUpdated().getTime())));
 
-        if (deviceConfigOptional.isPresent()) {
-            params.put(LAST_RETRIEVAL, String.valueOf(deviceConfigOptional.get().getLastUpdated().getTime()));
+        final String scriptFile = getKeyedString(parameters, SCRIPT_FILE, null);
+        try {
+            if (scriptFile != null) {
+                String script = parseScriptFile(scriptFile);
+                params.put(SCRIPT, script);
+            }
+        } catch (Exception e) {
+            LOG.error("Error while parsing script file {}", scriptFile);
         }
-
         return params;
     }
 
+    private static String parseScriptFile(String fileName) throws IOException {
+        String opennmsHome = ConfigFileConstants.getHome();
+        File opennmsHomeDir = new File(opennmsHome);
+        if (!opennmsHomeDir.exists()) {
+            LOG.debug("The specified home directory does not exist.");
+            throw new FileNotFoundException("The OpenNMS home directory \"" + opennmsHome + "\" does not exist.");
+        }
```

Comment:
This seems to be overkill to check. But you could use `Path` and `Files#exists` here to simplify.